invert-models/src/commonMain/kotlin/com/squareup/invert/models/ExtraMetadata.kt

```diff
@@ -11,4 +11,6 @@ data class ExtraMetadata(
   val key: ExtraKey,
   val type: ExtraDataType,
   val description: String,
+  // Whether this metadata field should be filterable in the code references view.
+  val filterable: Boolean = false,
```

Comment:
My main concern was potentially having extras that have too high of a cardinality for a drop down to be appropriate. If there's some other place to specify customizations for this other than the metadata itself, let me know. I don't love that this is included in the actual stat metadata as its very specific to the HTML UI

Reply:
This makes sense, but it also feels like a user will want to filter by something after the fact.

What about a button or something on the extras column header that you can click to filter by and then that populates a list.  `or` A "+" button where you can add a filter?

Sure, it can blow up if there are thousands, but there probably won't be.

That way we don't need this sort of control at the collection level.

Reply:
Hm.. yeah, I like the flexibility, but don't like the extra step / potential lack of discoverability. Perhaps a decent middle ground.

That goes a bit beyond what I currently know how to do in this codebase though - I'm not much of a web front end dev 😅

Reply:
Another alternative I thought of was to add a new `ExtraDataType` called `ENUM` thats the same as `STRING` but signals that its a set of known values. but thats also a bit weird - you'd end up marking things as `ENUM` just to get the filter even if it isn't a set of known values

Reply:
I noticed we already have drop downs with thousands of entries due to the number of modules in our codebase. I've updated this to instead just check for a hard-coded number of maximum possible values to enable filtering.